### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by viktoria-rybenchuk**

### CODE REVIEW

**Review by NazarHladaniuk — APPROVED** [review]

File: app/main.py [hunk 1]
```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
```
Comment (Serhii-Leonenko) [inline app/main.py:8]: Fix indentation here, move name to the new line to increase readability

File: app/main.py [hunk 2]
```diff
@@ -1 +1,31 @@
...
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Herbivore) and herbivore.hidden is False:
+            herbivore.health -= 50
+        if herbivore.health <= 0:
+            Animal.alive.remove(herbivore)
```
Comment (Serhii-Leonenko) [inline app/main.py:31]: The method should be static, self is redudant

File: app/main.py [hunk 3]
```diff
@@ -1 +1,31 @@
...
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        name = self.name
+        health = self.health
+        hidden = self.hidden
+        return f"{{Name: {name}, Health: {health}, Hidden: {hidden}}}"
```
Comment (Serhii-Leonenko) [inline app/main.py:18]: these variables are redundant in this case. Use brackets to wrap f-string

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -1 +1,32 @@
...
+    ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}," \
+               f" Hidden: {self.hidden}}}"
```
Comment (YuriiKindrat) [inline app/main.py:18]: try not to use " \ " here, use brackets to break the line

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]

**Review by YuriiKindrat — APPROVED** [review]

**Pull request merged**